src/Okta.Sdk.IntegrationTests/GroupScenarios.cs

```diff
@@ -507,5 +509,650 @@ public async Task RemoveGroupTargetFromGroupAdministratorRoleGivenToGroup()
                 await createdGroup3.DeleteAsync();
             }
         }
+
+        [Fact]
+        public async Task CreateGroupRule()
+        {
+            var testClient = TestClient.Create();
+            var guid = Guid.NewGuid();
+            var testNickName = $"dotnet-sdk-{nameof(CreateGroupRule)}";
+            var testUserName = $"{testNickName}-{guid}@example.com";
+
+            var profile = new UserProfile
+            {
+                FirstName = "John",
+                LastName = $"{nameof(CreateGroupRule)}",
+                Email = testUserName,
+                Login = testUserName,
+                ["nickName"] = testNickName,
+            };
+
+            var createdUser = await testClient.Users.CreateUserAsync(new CreateUserWithPasswordOptions
+            {
+                Profile = profile,
+                Password = "Abcd1234",
+            });
+
+            var createdGroup = await testClient.Groups.CreateGroupAsync(new CreateGroupOptions
+            {
+                Name = $"dotnet-sdk:{nameof(CreateGroupRule)}-{guid}",
+                Description = $"dotnet-sdk:{nameof(CreateGroupRule)}-{guid}",
+            });
+
+            Thread.Sleep(3000);
```

Comment:
Is this necessary? If so, let's leave a comment explaining why we need it.

Reply:
These are needed to avoid potentially trying to read data that hasn't been fully committed.  I'll add comments.